lib/imagec/imagec.go

```diff
@@ -289,13 +289,23 @@ func updateRepositoryCache(ic *ImageC) error {
 			return fmt.Errorf("ImageID not found by LayerID(%s) in RepositoryCache", imageLayerID)
 		}
 	}
-	// AddReference will add the tag / digest as appropriate and will persist
-	// to the portlayer k/v
+	// AddReference will add the repo:tag to the repositoryCache and save to the portLayer
 	err = repoCache.AddReference(ref, ic.ImageID, true, imageLayerID, true)
 	if err != nil {
 		return fmt.Errorf("Unable to Add Image Reference(%s): %s", ref.String(), err.Error())
 	}
 
+	dig, err := reference.ParseNamed(fmt.Sprintf("%s@%s", ref.Name(), ic.ManifestDigest))
+	if err != nil {
+		return fmt.Errorf("Unable to parse digest: %s", err.Error())
```

Comment:
Why do you add your own error here? Would not it be better just return original one?